bin/p2-fetch-manifest/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/manifest"
+	"github.com/square/p2/pkg/store/consul"
+	"github.com/square/p2/pkg/store/consul/consulutil"
+	"github.com/square/p2/pkg/store/consul/flags"
+	"github.com/square/p2/pkg/version"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+)
+
+const (
+	cmdFetchText = "fetch"
+)
+
+var (
+	cmdFetch     = kingpin.Command(cmdFetchText, "Fetch the manifest for a given pod from the datastore and write it to a file on the disk. Combines well with p2-launch.")
+	fetchTree    = cmdFetch.Flag("tree", "Intent or Reality").Default("intent").Short('t').Enum("intent", "reality")
+	fetchPod     = cmdFetch.Flag("podID", "The pod ID for which to fetch its manifest").Short('i').Required().String()
+	fetchEntropy = cmdFetch.Flag("force-restart", "Introduce entropy into the manifest before outputting it. Scheduling the resulting manifest will definitely cause a restart of the pod.").Short('p').Bool()
+)
+
+func main() {
+	kingpin.Version(version.VERSION)
+	logger := logging.NewLogger(logrus.Fields{})
+	cmd, consulOpts, labeler := flags.ParseWithConsulOptions()
+	_ = labeler
+	client := consul.NewConsulClient(consulOpts)
+
+	switch cmd {
+	case cmdFetchText:
+		hostname, err := os.Hostname()
+		if err != nil {
+			logger.Fatal(err)
+		}
+		man, err := fetch(client, *fetchTree, hostname, *fetchPod, *fetchEntropy)
+		if err != nil {
+			logger.Fatal(err)
+		}
+		filename := fmt.Sprintf("%s.yaml.signed", *fetchPod)
+		file, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE, 0600)
+		if err != nil {
+			logger.Fatalf("Could not open file at %s. %v", filename, err)
+		}
+
+		err = man.Write(file)
```

Comment:
the filename implies that we wish this to be signed. I believe this works fine if we do not modify the manifest, as the manifest comes out of the datastore signed.

What happens if we wish to introduce entropy? How do we sign the modified manifest?